lua/entities/ent_jack_gmod_ezmachine_base.lua

```diff
@@ -147,6 +147,20 @@ if(SERVER)then
 		JMod.Hint(ply, self.ClassName)
 		return ent
 	end
+	function ENT:Initialize()
+	if(self:CustomInit())then
+		self:CustomInit()
+	end
+	if(self:GetGrade())then
+		self:SetGrade(1)
+	end
+    self.Durability=self.MaxDurability
+	--
+    self.EZupgradable=true
+    self.UpgradeProgress={}
+    self.UpgradeCosts=JMod.CalculateUpgradeCosts(JMod.Config.Craftables[self.PrintName] and JMod.Config.Craftables[self.PrintName].craftingReqs)
```

Comment:
only do these two things if upgradadable is true

Reply:
Roger doger.